be/src/vec/functions/function_string.h

```diff
@@ -893,22 +893,21 @@ class FunctionStringConcatWs : public IFunction {
         const auto& src_array_offsets = array_column.get_offsets();
         size_t current_src_array_offset = 0;
 
+        auto& sep_offsets = *offsets_list[0];
+        auto& sep_chars = *chars_list[0];
+        auto& sep_nullmap = *null_list[0];
+        int sep_size = sep_offsets[0];
+        const char* sep_data = reinterpret_cast<const char*>(&sep_chars[0]);
+        std::string_view sep(sep_data, sep_size);
+
```

Comment:
If it is moved outside the loop, it will always use the data in the first row as the separator?

Reply:
Your are right, finally I find the real reason of overflow is version() not process input_rows_count rightly.